addons/mass_mailing/models/mailing.py

```diff
@@ -1281,6 +1366,43 @@ def _get_default_mailing_domain(self):
 
         return mailing_domain
 
+    def _get_image_by_url(self, url, session):
+        maxsize = int(tools.config.get("import_image_maxbytes", DEFAULT_IMAGE_MAXBYTES))
+        _logger.debug("Trying to import image from URL: %s", url)
+        try:
+            response = session.get(url, timeout=int(tools.config.get("import_image_timeout", DEFAULT_IMAGE_TIMEOUT)))
+            response.raise_for_status()
+
+            if response.headers.get('Content-Length') and int(response.headers['Content-Length']) > maxsize:
+                raise ImportValidationError(
+                    _("File size exceeds configured maximum (%s bytes)", maxsize)
+                )
+
+            content = bytearray()
+            for chunk in response.iter_content(DEFAULT_IMAGE_CHUNK_SIZE):
+                content += chunk
+                if len(content) > maxsize:
+                    raise ImportValidationError(
+                        _("File size exceeds configured maximum (%s bytes)", maxsize)
+                    )
+
+            image = Image.open(io.BytesIO(content))
+            w, h = image.size
+            if w * h > 42e6:  # Nokia Lumia 1020 photo resolution
```

Comment:
Ok for copy pasting, but nokia lumia 1020 dates from 2013 ... and anyway do we want 42Mpixels image, isn't that a bit too much anyway N

Reply:
What would you suggest?

Personally I'm not sure the moment we're saving the e-mail the use created is the best moment to decide the resolution they picked was too big honestly (we may want something like that in the media dialog though if it doesn't exist yet), and in this specific case we use this only to crop background images for Outlook and we'll just ignore it if it fails, so adding cases where it might fail seems counterproductive to me in this particular instance.

Reply:
Well initially my comment was that lumia 1020 comment is probably out of scope... Not sure it is still used anywhere in the world :D .

For the size, don't have strong opinions ... all images should be resized anyway (not sure if it is done or not, did not play with it since some time to be honest ^^ ). But that's another issue, probably.

Reply:
I'll remove the comment :-P

Reply:
MOAR PIXELS !!!